File: basic-algorithms/fizzBuzz/fizz-buzz.go
package main

import (
	"fmt"
	"strconv"
)

func fizzBuzz(n int) []string {
	result := make([]string, 0, n)
	for i := 1; i <= n; i++ {
		switch {
			case i%3 == 0 && i%5 == 0:
				result = append(result, "FizzBuzz")
			case i%3 == 0:
				result = append(result, "Fizz")
			case i%5 == 0:
				result = append(result, "Buzz")
			default:
				result = append(result, strconv.Itoa(i))
		}
	}

	return result
}

func main() {
	fmt.Println(fizzBuzz(15))
	fmt.Println(fizzBuzz(5))
}

File: fizzbuzz/ts/src/fizzbuzz.ts
export function fizzbuzz(n: number): void {
  if (n <= 0) {
    throw new Error("Input must be a positive integer.");
  }

  for(let i = 1; i <= n; i++) {
    console.log(`${i}: ${printFizzBuzz(i)}`);
  }
}

function printFizzBuzz(n: number): string {
    let result = '';

    if (n % 3 === 0) {
        result += 'Fizz';
    }

    if (n % 5 === 0) {
        result += 'Buzz';
    }

    if (n % 3 === 0 && n % 5 === 0) {
        result = 'FizzBuzz';
    }

    return result || n.toString();
}